app/page/itinerary.cjsx

```diff
@@ -34,11 +98,64 @@ class ItineraryPage extends React.Component
         {name: 'description', content: @context.intl.formatMessage {id: 'itinerary-page.description', defaultMessage: "Route"}}
       ]
 
+    if @state.fullscreen
+      content =
+        <div style={"height": "100%"} onTouchStart={(e) => e.stopPropagation()} onMouseDown={(e) => e.stopPropagation()}>
+          <Map
+            ref="map2"
+            className="fullscreen"
+            leafletObjs={leafletObj}
+            lat={if @state.lat then @state.lat else itineraries[parseInt(@props.params.hash)].legs[0].from.lat}
+            lon={if @state.lon then @state.lon else itineraries[parseInt(@props.params.hash)].legs[0].from.lon}
+            zoom=16
+            fitBounds={false}>
+            <div className="fullscreen-toggle" onClick={@toggleFullscreenMap}>
+              <Icon img={'icon-icon_maximize'} className="cursor-pointer" />
+            </div>
+          </Map>
+        </div>
+
+    else
+      content =
+        <div style={height: "100%"}>
+          <div onTouchStart={(e) => e.stopPropagation()} onMouseDown={(e) => e.stopPropagation()}>
+            <Map
+              ref="map"
+              leafletObjs={leafletObj}
+              lat={if @state.lat then @state.lat else itineraries[parseInt(@props.params.hash)].legs[0].from.lat}
+              lon={if @state.lon then @state.lon else itineraries[parseInt(@props.params.hash)].legs[0].from.lon}
+              zoom=16
+              fitBounds={false}
+              leafletOptions={dragging: false, touchZoom: false, scrollWheelZoom: false, doubleClickZoom: false, boxZoom: false}>
+              <div className="fullscreen-toggle" onClick={@toggleFullscreenMap}>
+                <Icon img={'icon-icon_maximize'} className="cursor-pointer" />
+              </div>
+            </Map>
+          </div>
+          <SwipeableViews
+            index={parseInt(@props.params.hash)}
+            className="itinerary-swipe-views-root"
+            slideStyle={{height: "100%"}}
+            containerStyle={{height: "100%"}}
+            onChangeIndex={@switchSlide}>
+            {@getSlides(itineraries)}
+          </SwipeableViews>
+          <div className="itinerary-tabs-container">
+            <Tabs
+              onChange={() => return}
```

Comment:
This prop function was required in order to the Tabs component to update itself when slide occurs in desktop display mode. In real mobile device, the tab dots updated correctly even without this, but on desktop google chrome display mode, they did not unless this prop function. In material-ui example, there was also two change listeners, one in tabs and one in the react swipeable views: http://www.material-ui.com/#/components/tabs (swipeable example)